examples/guided.py

```diff
@@ -410,8 +408,6 @@ def perform_installation(mountpoint):
 	# Temporarily disabling keep_partitions if config file is loaded
 	archinstall.arguments['harddrive'].keep_partitions = False
 	# Temporary workaround to make Desktop Environments work
-	archinstall.storage['_desktop_profile'] = archinstall.arguments.get('desktop', None)
-	if archinstall.arguments.get('profile', None):
-		archinstall.arguments['profile'] = Profile(installer=None, path=archinstall.arguments['profile']['path'])
+	archinstall.arguments['profile'] = archinstall.Profile(None, archinstall.arguments.get('desktop', None))
```

Comment:
Wouldn't this overwrite any `--desktop` argument from line #L181?

Reply:
#L181 will be unreachable if config is passed, as ask_user_questions won't be called

Reply:
I meant to write `--profile`.
Lets say you do `python -m archinstall guided --profile awesome`, on #L183 that would have resulted in the profile being sourced in `archinstall.arguments['profile']`, but now that will be overwritten by potentially something in `--desktop`.

Scenario:
```
python -m archinstall guided --profile=minimal
```
Initially, that would be:
```python
archinstall.arguments['profile'] == archinstall.Profile(None, 'minimal')
```
But this change *(I think)*  would mean that `archinstall.arguments['profile'] == None` because `archinstall.arguments.get('desktop', None)` was never set so it will inject `None` into `archinstall.arguments['profile']` on #L411?

Reply:
Okay, I understand now, but as per my understanding this line would be only triggered if and only if config file is loaded, otherwise this would be unreachable. Please correct me if I'm wrong.

Reply:
`profile` key will be ignored if present in config file

Reply:
Ah you're right, my bad! That piece of logic is still a bit new to me.

So final question from my end would be why we're putting `desktop` argument into `profile` argument?
This will cause this to be triggered every time: https://github.com/archlinux/archinstall/blob/bbb4599165a644bbd81b085fb3210cd0e497d503/profiles/desktop.py#L51-L52

This is problem an issue created outside of this PR, but I noticed it here.
The issue of #486 is probably more to do with this not being converted to an actual profile object: https://github.com/archlinux/archinstall/blob/c2f3b9e2088bfd5ed1c6a5f80bab28277653bb8a/examples/guided.py#L415

```python
if archinstall.arguments.get('profile', None):
	archinstall.arguments['profile'] = archinstall.list_profiles()[archinstall.arguments['profile']]
	archinstall.arguments['profile'] = Profile(installer=None, path=archinstall.arguments['profile']['path'])
```
This would solve it.
Since in `ask_user_questions()` this is handled, but in the config loading it isn't.